app/decorators/marriage_decorator.rb

```diff
@@ -5,6 +5,8 @@ class MarriageDecorator < Draper::Decorator
 
   memo_wise \
   def other_partner
-    partners.where.not(id: h.current_user).first
+    partner_ids = [partner_1_id, partner_2_id]
+    partner_id = (partner_ids - [h.current_user.id]).first
+    User.find(partner_id)
```

Comment:
(Mostly unrelated change to try to optimize this query a bit (inspired by looking at our SQL queries more closely).)